command/host_create.go

```diff
@@ -61,8 +66,70 @@ func (c *createHost) Execute(ctx context.Context, comm client.Communicator,
 			return err
 		}
 	}
+	startTime := time.Now()
+	ids, err := comm.CreateHost(ctx, taskData, *c.CreateHost)
+	if err != nil || c.CreateHost.CloudProvider != apimodels.ProviderDocker || c.CreateHost.Background == false {
```

Comment:
I believe you're handling the background parameter incorrectly here - if background is false, we should continue on and do what's below in the same thread as what's above. If background is true, then we do what's below in a separate thread